src/argilla/server/schemas/v0/users.py

```diff
@@ -0,0 +1,64 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from datetime import datetime
+from typing import Any, List, Optional
+from uuid import UUID
+
+from argilla.server.constants import ES_INDEX_REGEX_PATTERN
+from argilla.server.enums import UserRole
+from argilla.server.pydantic_v1 import BaseModel, Field, constr
+from argilla.server.pydantic_v1.utils import GetterDict
+
+USER_USERNAME_REGEX = ES_INDEX_REGEX_PATTERN
+USER_PASSWORD_MIN_LENGTH = 8
+USER_PASSWORD_MAX_LENGTH = 100
+
+
+class UserCreate(BaseModel):
+    first_name: constr(min_length=1, strip_whitespace=True)
+    last_name: Optional[constr(min_length=1, strip_whitespace=True)]
+    username: str = Field(regex=USER_USERNAME_REGEX, min_length=1)
+    role: Optional[UserRole]
+    password: constr(min_length=USER_PASSWORD_MIN_LENGTH, max_length=USER_PASSWORD_MAX_LENGTH)
+    workspaces: Optional[List[str]]
```

Comment:
In the refactor that I did for API v1 schemas I'm trying to define first when possible the canonical entity (in this case the `User`) and later schemas that create/modify (like `UserCreate` in this case). So my suggestion is moving this schema class to the end of the file.